### PR TITLE
#3043 - create table - update to oracle 21

### PR DESCRIPTION
**Pull request by mlorek**
create table stmt update to oracle 21

### CODE REVIEW

File: sql/plsql/PlSqlParser.g4 [hunk 1]
```diff
@@ -1993,15 +1993,15 @@ blockchain_row_retention_clause
     ;
 
 blockchain_hash_and_data_format_clause
-    : HASHING USING sha2_512 VERSION v1
+    : HASHING USING sha2_512_quoted VERSION v1_quoted
```
Comment (KvanTTT) [inline sql/plsql/PlSqlParser.g4:1996]: Can `sha2_512_quoted` and `v1_quoted` just be inlined? What is the purpose of primitive parser rules?

```suggestion
    : HASHING USING SHA2_512_Q VERSION V1_Q
```

**Comment by mlorek** [comment]
@KvanTTT are you ok with the changes?

**Review by KvanTTT — CHANGES_REQUESTED** [review]
Sorry, but could you split tokens reordering (for instance, I see `A_LETTER` just was moved below) and feature changes? It's hard for reviewing when they are mixed.

File: sql/plsql/PlSqlParser.g4 [hunk 2]
```diff
@@ -3090,19 +3309,61 @@ end_time_column
...
     : GENERATED (ALWAYS | BY DEFAULT (ON NULL_)?)? AS IDENTITY ( '(' (sequence_start_clause | sequence_spec)* ')' )?
     ;
 
+evaluation_edition_clause
+    : EVALUATE USING (CURRENT EDITION | EDITION edition_name | NULL_ EDITION)
+    ;
+
+unusable_edition_clause
+    : (UNUSABLE BEFORE (CURRENT EDITION | EDITION edition_name))?
+        (UNUSABLE BEGINNING WITH (CURRENT EDITION | EDITION edition_name | NULL_ EDITION) )?
```
Comment (KvanTTT) [inline sql/plsql/PlSqlParser.g4:3364]:
```suggestion
        (UNUSABLE BEGINNING WITH ((CURRENT | NULL_) EDITION | EDITION edition_name) )?
```

File: sql/plsql/PlSqlParser.g4 [hunk 3]
```diff
@@ -3090,19 +3309,61 @@ end_time_column
...
 
 virtual_column_definition
-    : column_name datatype? autogenerated_sequence_definition?
-        VIRTUAL? inline_constraint*
+    : column_name (datatype COLLATE column_collation_name)?
+        (VISIBLE | INVISIBLE)?
+        autogenerated_sequence_definition?
+        VIRTUAL?
+        evaluation_edition_clause?
+        unusable_edition_clause
```
Comment (KvanTTT) [inline sql/plsql/PlSqlParser.g4:3350]: I suggest changing it to optional:

```suggestion
        unusable_edition_clause?
```

And making `unusable_edition_clause` not empty.

File: sql/plsql/PlSqlParser.g4 [hunk 4]
```diff
@@ -3090,19 +3309,61 @@ end_time_column
...
+        inline_constraint*
     ;
 
 autogenerated_sequence_definition
     : GENERATED (ALWAYS | BY DEFAULT (ON NULL_)?)? AS IDENTITY ( '(' (sequence_start_clause | sequence_spec)* ')' )?
     ;
 
+evaluation_edition_clause
+    : EVALUATE USING (CURRENT EDITION | EDITION edition_name | NULL_ EDITION)
```
Comment (KvanTTT) [inline sql/plsql/PlSqlParser.g4:3359]:
```suggestion
    : EVALUATE USING ((CURRENT | NULL_) EDITION | EDITION edition_name)
```

**Review by KvanTTT — CHANGES_REQUESTED** [review]

File: sql/plsql/PlSqlParser.g4 [hunk 5]
```diff
@@ -6614,6 +6872,7 @@ non_reserved_keywords_pre12c
     | SWITCHOVER
     | SYNCHRONOUS
     | SYNC
+    | SYS
```
Comment (KvanTTT) [inline sql/plsql/PlSqlParser.g4:6875]: Not only `MEMOPTIMIZE` and `SYS`, but all new tokens from this pull request. They can be identifiers as well.

**Review by KvanTTT — CHANGES_REQUESTED** [review]

**Review by KvanTTT — APPROVED** [review]

**Pull request merged**